### PR TITLE
Dewey: Converting to structured answer

### PR DESCRIPTION
**Pull request by mintsoft**
Finishing off: https://github.com/duckduckgo/zeroclickinfo-goodies/pull/3213

###### People to notify (@mention interested parties)

@AlterationBrick 

---

Instant Answer Page: https://duck.co/ia/view/dewey

[Maintainer](http://docs.duckduckhack.com/maintaining/guidelines.html): @navneetzz

### CODE REVIEW

**Comment by daxtheduck** [comment]
Metadata placeholder post
            
*If you haven't already, please edit your first comment to include a link to the IA page at the bottom.*
            
---
            
**How to find your IA page**
- New Instant Answer
  -- follow the steps [here](https://duck.co/ia/new_ia) to create your IA page.
              
- Existing Instant Answer
  -- find your IA page from this [list](https://duck.co/ia)

File: lib/DDG/Goodie/Dewey.pm [hunk 1]
```diff
@@ -5,84 +5,88 @@ use strict;
...
-
-sub single_format {
-    "$_[0] is $_[1] in the Dewey Decimal System.";
+    if(exists($nums{"$num\n"})) {
+        $data->{$num} = get_info($num) or return;
+    }
 }
 
 handle remainder => sub {
     return unless s/^(?:the)?\s*(?:decimal)?\s*(?:system)?\s*(?:numbers?|\#)?\s*
```
Comment (mintsoft) [inline lib/DDG/Goodie/Dewey.pm]: If I replace this with a non- `s///iex` version this breaks the IA completely; I can't see any dependency on `$_` anywhere left; if someone could point it out I'd be much appreciated!

**Comment by mintsoft** [comment]
This now works and the tests pass etc; although I dislike the implemetation I think it's very .. unclear at the moment especially with the `s///` in the guard

**Comment by GuiltyDolphin** [comment]
@mintsoft Would you mind squashing some of those commits? The history is a nightmare to read on this one.

**Comment by mintsoft** [comment]
@GuiltyDolphin Obviously my attempt to tidy up broke the branch completely (thanks git!); however it should be cleaned now

**Comment by GuiltyDolphin** [comment]
@mintsoft Thanks.

@moollaza Can we get this deployed?

**Comment by moollaza** [comment]
Deploying...

**Comment by moollaza** [comment]
@mintsoft Thanks! This LGTM :+1:

**Comment by mintsoft** [comment]
Thanks @AlterationBrick for doing all the hard work!

**Pull request merged**